From 23b4b89defb3a1e3f73eac2eaf7d03da1352f6da Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@nxp.com>
Date: Wed, 17 Feb 2016 16:35:28 +0200
Subject: [PATCH 1170/1429] dpa_offload: Add example DTS files for P4080DS

Added the following offloading scenarios DTS files for the
P4080DS platform:

	* USDPAA apps scenario DTS files
	* IP Fragmentation scenario DTS files
	* IP Reassembly scenario DTS files
	* Ethernet-to-Ethernet IPSec offloading scenario DTS files

Signed-off-by: Marian Chereji <marian.chereji@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 ...ared-interfaces-for-ipsec-multiple-intances.dts |  218 ++++++++++++++++++++
 .../dts/p4080ds-usdpaa-shared-interfaces.dts       |  191 +++++++++++++++++
 .../staging/fsl_dpa_offload/dts/p4080ds-usdpaa.dts |  176 ++++++++++++++++
 .../fsl_dpa_offload/dts/p4080si-chosen-offld.dtsi  |   21 ++
 .../staging/fsl_dpa_offload/dts/p4080si-pre.dtsi   |  169 +++++++++++++++
 5 files changed, 775 insertions(+), 0 deletions(-)
 create mode 100644 drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces-for-ipsec-multiple-intances.dts
 create mode 100644 drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces.dts
 create mode 100644 drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa.dts
 create mode 100644 drivers/staging/fsl_dpa_offload/dts/p4080si-chosen-offld.dtsi
 create mode 100644 drivers/staging/fsl_dpa_offload/dts/p4080si-pre.dtsi

diff --git a/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces-for-ipsec-multiple-intances.dts b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces-for-ipsec-multiple-intances.dts
new file mode 100644
index 0000000..95dc9c6
--- /dev/null
+++ b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces-for-ipsec-multiple-intances.dts
@@ -0,0 +1,218 @@
+/*
+ * P4080DS Device Tree Source
+ *
+ * Copyright 2009-2014 Freescale Semiconductor Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/include/ "p4080ds.dts"
+
+/ {
+	/* NB: "bpool-ethernet-seeds" is not set to avoid buffer seeding,
+	 * because apps seed these pools with buffers allocated at
+	 * run-time.
+	 * HOWEVER, the kernel driver requires the buffer-size so
+	 * "fsl,bpool-ethernet-cfg" is set. It also mis-interprets
+	 * things if the base-address is zero (hence the 0xdeadbeef
+	 * values).
+	 */
+	bp7: buffer-pool@7 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <7>;
+		fsl,bpool-ethernet-cfg = <0 0 0 192 0 0xdeadbeef>;
+		fsl,bpool-thresholds = <0x400 0xc00 0x0 0x0>;
+	};
+	bp8: buffer-pool@8 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <8>;
+		fsl,bpool-ethernet-cfg = <0 0 0 576 0 0xabbaf00d>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp9: buffer-pool@9 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <9>;
+		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp16: buffer-pool@16 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <16>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp17: buffer-pool@17 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <17>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	fsl,dpaa {
+		ethernet@0 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x50 1 0x51 1>;
+			fsl,qman-frame-queues-tx = <0x70 1 0x71 1>;
+		};
+
+	        /* this inteface should be init by usdpaa */
+	        ethernet@1 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x48 1 0x49 1>;
+			fsl,qman-frame-queues-tx = <0x68 1 0x69 1>;
+	        };
+
+		ethernet@2 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x54 1 0x55 1>;
+			fsl,qman-frame-queues-tx = <0x74 1 0x75 1>;
+		};
+		ethernet@3 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x56 1 0x57 1>;
+			fsl,qman-frame-queues-tx = <0x76 1 0x77 1>;
+		};
+		ethernet@4 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5a 1 0x5b 1>;
+			fsl,qman-frame-queues-tx = <0x7a 1 0x7b 1>;
+		};
+		ethernet@5 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5c 1 0x5d 1>;
+			fsl,qman-frame-queues-tx = <0x7c 1 0x7d 1>;
+		};
+		ethernet@6 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5e 1 0x5f 1>;
+			fsl,qman-frame-queues-tx = <0x7e 1 0x7f 1>;
+		};
+		ethernet@7 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x60 1 0x61 1>;
+			fsl,qman-frame-queues-tx = <0x80 1 0x81 1>;
+		};
+		ethernet@8 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x62 1 0x63 1>;
+			fsl,qman-frame-queues-tx = <0x82 1 0x83 1>;
+		};
+		/* ethernet@9 declared as shared MAC. USDPAA will seed buffers to
+		 * this buffer pool. The ethernet driver will initialize the RX default,
+		 * RX error, TX error, TX confirm and 8 TX Frame queues. On receiving frame
+		 * at this interface, the ethernet driver will do kmap_atomic/kunmap_atomic
+		 * for that frame. */
+		ethernet@9 {
+			compatible = "fsl,p4080-dpa-ethernet-shared", "fsl,dpa-ethernet-shared";
+			fsl,bman-buffer-pools = <&bp17>;
+			fsl,qman-frame-queues-rx = <0x66 1 0x67 1 0x2000 3>;
+			fsl,qman-frame-queues-tx = <0 1 0 1 0x3000 8>;
+		};
+		/* ethernet@16 declared as MAC-less interface with no "fsl,fman-mac" property.
+		 * USDPAA will seed buffers to this buffer pool and initialize 8 TX Frame
+		 * queues. The ethernet driver will initialize 8 RX default Frame queues.
+		 * On receiving frame at this interface, the ethernet driver will do
+		 * kmap_atomic/kunmap_atomic for that frame. */
+		ethernet@16 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <4000 8>;
+			fsl,qman-frame-queues-tx = <4008 8>;
+			local-mac-address = [00 11 22 33 44 55];
+		};
+		ethernet@17 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <5000 8>;
+			fsl,qman-frame-queues-tx = <5008 8>;
+			local-mac-address = [00 11 22 33 44 66];
+		};
+		ethernet@18 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <6000 8>;
+			fsl,qman-frame-queues-tx = <6008 8>;
+			local-mac-address = [00 11 22 33 44 77];
+		};
+		ethernet@19 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <7000 8>;
+			fsl,qman-frame-queues-tx = <7008 8>;
+			local-mac-address = [00 11 22 33 44 88];
+		};
+
+		dpa_fman1_oh1: dpa-fman1-oh@1 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x64 1 0x65 1>;
+			fsl,fman-oh-port = <&fman1_oh1>;
+		};
+		dpa_fman1_oh2: dpa-fman1-oh@2 { /* Define frame queues for the OH port*/
+			compatible = "fsl,dpa-oh"; /* <OH Rx error, OH Rx default> */
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
+			fsl,bman-buffer-pools = <&bp8>;
+			fsl,fman-oh-port = <&fman1_oh2>;
+		};
+		dpa_fman1_oh3: dpa-fman1-oh@3 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x70 1 0x71 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman1_oh3>;
+		};
+		dpa_fman0_oh1: dpa-fman0-oh@1 {
+			compatible = "fsl,dpa-oh";
+			fsl,qman-frame-queues-oh = <0x77 1 0x78 1>;
+			fsl,fman-oh-port = <&fman0_oh1>;
+		};
+		dpa_fman0_oh2: dpa-fman0-oh@2 {
+			compatible = "fsl,dpa-oh";
+			fsl,qman-frame-queues-oh = <0x79 1 0x80 1>;
+			fsl,fman-oh-port = <&fman0_oh2>;
+		};
+		dpa_fman0_oh3: dpa-fman0-oh@3 {
+			compatible = "fsl,dpa-oh";
+			fsl,qman-frame-queues-oh = <0x81 1 0x82 1>;
+			fsl,fman-oh-port = <&fman0_oh3>;
+		};
+	};
+};
diff --git a/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces.dts b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces.dts
new file mode 100644
index 0000000..1486ceb
--- /dev/null
+++ b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa-shared-interfaces.dts
@@ -0,0 +1,191 @@
+/*
+ * P4080DS Device Tree Source
+ *
+ * Copyright 2009-2012 Freescale Semiconductor Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/include/ "p4080ds.dts"
+
+/ {
+	/* NB: "bpool-ethernet-seeds" is not set to avoid buffer seeding,
+	 * because apps seed these pools with buffers allocated at
+	 * run-time.
+	 * HOWEVER, the kernel driver requires the buffer-size so
+	 * "fsl,bpool-ethernet-cfg" is set. It also mis-interprets
+	 * things if the base-address is zero (hence the 0xdeadbeef
+	 * values).
+	 */
+	bp7: buffer-pool@7 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <7>;
+		fsl,bpool-ethernet-cfg = <0 0 0 192 0 0xdeadbeef>;
+		fsl,bpool-thresholds = <0x400 0xc00 0x0 0x0>;
+	};
+	bp8: buffer-pool@8 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <8>;
+		fsl,bpool-ethernet-cfg = <0 0 0 576 0 0xabbaf00d>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp9: buffer-pool@9 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <9>;
+		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp16: buffer-pool@16 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <16>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp17: buffer-pool@17 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <17>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	fsl,dpaa {
+		ethernet@0 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x50 1 0x51 1>;
+			fsl,qman-frame-queues-tx = <0x70 1 0x71 1>;
+		};
+
+		/* ethernet@1 will be used as a normal Linux ethernet that
+		 * interfaces to the kernel network stack. All others will be
+		 * dedicated for use by usdpaa */
+
+		ethernet@2 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x54 1 0x55 1>;
+			fsl,qman-frame-queues-tx = <0x74 1 0x75 1>;
+		};
+		ethernet@3 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x56 1 0x57 1>;
+			fsl,qman-frame-queues-tx = <0x76 1 0x77 1>;
+		};
+		ethernet@4 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5a 1 0x5b 1>;
+			fsl,qman-frame-queues-tx = <0x7a 1 0x7b 1>;
+		};
+		ethernet@5 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5c 1 0x5d 1>;
+			fsl,qman-frame-queues-tx = <0x7c 1 0x7d 1>;
+		};
+		ethernet@6 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x5e 1 0x5f 1>;
+			fsl,qman-frame-queues-tx = <0x7e 1 0x7f 1>;
+		};
+		ethernet@7 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x60 1 0x61 1>;
+			fsl,qman-frame-queues-tx = <0x80 1 0x81 1>;
+		};
+		ethernet@8 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <0x62 1 0x63 1>;
+			fsl,qman-frame-queues-tx = <0x82 1 0x83 1>;
+		};
+		/* ethernet@9 declared as shared MAC. USDPAA will seed buffers to
+		 * this buffer pool. The ethernet driver will initialize the RX default,
+		 * RX error, TX error, TX confirm and 8 TX Frame queues. On receiving frame
+		 * at this interface, the ethernet driver will do kmap_atomic/kunmap_atomic
+		 * for that frame. */
+		ethernet@9 {
+			compatible = "fsl,p4080-dpa-ethernet-shared", "fsl,dpa-ethernet-shared";
+			fsl,bman-buffer-pools = <&bp17>;
+			fsl,qman-frame-queues-rx = <0x66 1 0x67 1 0x2000 3>;
+			fsl,qman-frame-queues-tx = <0 1 0 1 0x3000 8>;
+		};
+		/* ethernet@16 declared as MAC-less interface with no "fsl,fman-mac" property.
+		 * USDPAA will seed buffers to this buffer pool and initialize 8 TX Frame
+		 * queues. The ethernet driver will initialize 8 RX default Frame queues.
+		 * On receiving frame at this interface, the ethernet driver will do
+		 * kmap_atomic/kunmap_atomic for that frame. */
+		ethernet@16 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <4000 8>;
+			fsl,qman-frame-queues-tx = <4008 8>;
+			local-mac-address = [00 11 22 33 44 55];
+		};
+		ethernet@17 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <5000 8>;
+			fsl,qman-frame-queues-tx = <5008 8>;
+			local-mac-address = [00 11 22 33 44 66];
+		};
+		ethernet@18 {
+                        compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+                        fsl,bman-buffer-pools = <&bp16>;
+                        fsl,qman-frame-queues-rx = <6000 8>;
+                        fsl,qman-frame-queues-tx = <6008 8>;
+                        local-mac-address = [00 11 22 33 44 77];
+                };
+		dpa_fman1_oh1: dpa-fman1-oh@1 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x64 1 0x65 1>;
+			fsl,fman-oh-port = <&fman1_oh1>;
+		};
+		dpa_fman1_oh2: dpa-fman1-oh@2 { /* Define frame queues for the OH port*/
+			compatible = "fsl,dpa-oh"; /* <OH Rx error, OH Rx default> */
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
+			fsl,bman-buffer-pools = <&bp8>;
+			fsl,fman-oh-port = <&fman1_oh2>;
+		};
+		dpa_fman1_oh3: dpa-fman1-oh@3 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x70 1 0x71 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman1_oh3>;
+		};
+	};
+};
diff --git a/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa.dts b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa.dts
new file mode 100644
index 0000000..5f889e3
--- /dev/null
+++ b/drivers/staging/fsl_dpa_offload/dts/p4080ds-usdpaa.dts
@@ -0,0 +1,176 @@
+/*
+ * P4080DS Device Tree Source
+ *
+ * Copyright 2009-2012 Freescale Semiconductor Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/include/ "p4080ds.dts"
+
+/ {
+	/* NB: "bpool-ethernet-seeds" is not set to avoid buffer seeding,
+	 * because apps seed these pools with buffers allocated at
+	 * run-time.
+	 * HOWEVER, the kernel driver requires the buffer-size so
+	 * "fsl,bpool-ethernet-cfg" is set. It also mis-interprets
+	 * things if the base-address is zero (hence the 0xdeadbeef
+	 * values).
+	 */
+	bp7: buffer-pool@7 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <7>;
+		fsl,bpool-ethernet-cfg = <0 0 0 192 0 0xdeadbeef>;
+		fsl,bpool-thresholds = <0x400 0xc00 0x0 0x0>;
+	};
+	bp8: buffer-pool@8 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <8>;
+		fsl,bpool-ethernet-cfg = <0 0 0 576 0 0xabbaf00d>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp9: buffer-pool@9 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <9>;
+		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp16: buffer-pool@16 {
+		compatible = "fsl,p4080-bpool", "fsl,bpool";
+		fsl,bpid = <16>;
+		fsl,bpool-ethernet-cfg = <0 2048 0 1728 0 0>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+
+	fsl,dpaa {
+		ethernet@0 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x50 1 0x51 1>;
+			fsl,qman-frame-queues-tx = <0x70 1 0x71 1>;
+		};
+
+		/* ethernet@1 will be used as a normal Linux ethernet that
+		 * interfaces to the kernel network stack. All others will be
+		 * dedicated for use by usdpaa */
+
+		ethernet@2 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x54 1 0x55 1>;
+			fsl,qman-frame-queues-tx = <0x74 1 0x75 1>;
+		};
+		ethernet@3 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x56 1 0x57 1>;
+			fsl,qman-frame-queues-tx = <0x76 1 0x77 1>;
+		};
+		ethernet@4 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x5a 1 0x5b 1>;
+			fsl,qman-frame-queues-tx = <0x7a 1 0x7b 1>;
+		};
+		ethernet@5 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x5c 1 0x5d 1>;
+			fsl,qman-frame-queues-tx = <0x7c 1 0x7d 1>;
+		};
+		ethernet@6 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x5e 1 0x5f 1>;
+			fsl,qman-frame-queues-tx = <0x7e 1 0x7f 1>;
+		};
+		ethernet@7 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x60 1 0x61 1>;
+			fsl,qman-frame-queues-tx = <0x80 1 0x81 1>;
+		};
+		ethernet@8 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x62 1 0x63 1>;
+			fsl,qman-frame-queues-tx = <0x82 1 0x83 1>;
+		};
+		ethernet@9 {
+			compatible = "fsl,p4080-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x66 1 0x67 1>;
+			fsl,qman-frame-queues-tx = <0x86 1 0x87 1>;
+		};
+
+		/* ethernet@16 declared as MAC-less interface with no "fsl,fman-mac" property.
+		 * USDPAA will seed buffers to this buffer pool and initialize 8 TX Frame
+		 * queues. The ethernet driver will initialize 8 RX default Frame queues.
+		 * On receiving frame at this interface, the ethernet driver will do
+		 * kmap_atomic/kunmap_atomic for that frame. */
+		ethernet@16 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <4000 8>;
+			fsl,qman-frame-queues-tx = <4008 8>;
+			local-mac-address = [00 11 22 33 44 55];
+		};
+		ethernet@17 {
+			compatible = "fsl,p4080-dpa-ethernet-macless", "fsl,dpa-ethernet-macless";
+			fsl,bman-buffer-pools = <&bp16>;
+			fsl,qman-frame-queues-rx = <5000 8>;
+			fsl,qman-frame-queues-tx = <5008 8>;
+			local-mac-address = [00 11 22 33 44 66];
+		};
+
+		dpa_fman1_oh1: dpa-fman1-oh@1 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x64 1 0x65 1>;
+			fsl,fman-oh-port = <&fman1_oh1>;
+		};
+		dpa_fman1_oh2: dpa-fman1-oh@2 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman1_oh2>;
+		};
+		dpa_fman1_oh3: dpa-fman1-oh@3 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x70 1 0x71 1>;
+			fsl,bman-buffer-pools = <&bp9>;
+			fsl,fman-oh-port = <&fman1_oh3>;
+		};
+	};
+};
diff --git a/drivers/staging/fsl_dpa_offload/dts/p4080si-chosen-offld.dtsi b/drivers/staging/fsl_dpa_offload/dts/p4080si-chosen-offld.dtsi
new file mode 100644
index 0000000..1b9f530
--- /dev/null
+++ b/drivers/staging/fsl_dpa_offload/dts/p4080si-chosen-offld.dtsi
@@ -0,0 +1,21 @@
+chosen {
+	name = "chosen";
+
+	dpaa-extended-args {
+		fman1-extd-args {
+			cell-index = <1>;
+			compatible = "fsl,fman-extended-args";
+			dma-aid-mode = "port";
+			fman1_oh2-extd-args {
+				cell-index = <2>;
+				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
+				/* Define buffer layout parameters. Can be used
+				 * in fragmentation or header manip operations
+				 */
+				/* <manip extra space, data alignment> */
+				buffer-layout = <128 64>;
+			};
+		};
+	};
+};
diff --git a/drivers/staging/fsl_dpa_offload/dts/p4080si-pre.dtsi b/drivers/staging/fsl_dpa_offload/dts/p4080si-pre.dtsi
new file mode 100644
index 0000000..9d9625b
--- /dev/null
+++ b/drivers/staging/fsl_dpa_offload/dts/p4080si-pre.dtsi
@@ -0,0 +1,169 @@
+/*
+ * P4080/P4040 Silicon/SoC Device Tree Source (pre include)
+ *
+ * Copyright 2011 Freescale Semiconductor Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *	 notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *	 notice, this list of conditions and the following disclaimer in the
+ *	 documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *	 names of its contributors may be used to endorse or promote products
+ *	 derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/dts-v1/;
+
+/include/ "e500mc_power_isa.dtsi"
+
+/ {
+	compatible = "fsl,P4080";
+	#address-cells = <2>;
+	#size-cells = <2>;
+	interrupt-parent = <&mpic>;
+
+	aliases {
+		ccsr = &soc;
+		dcsr = &dcsr;
+
+		serial0 = &serial0;
+		serial1 = &serial1;
+		serial2 = &serial2;
+		serial3 = &serial3;
+		pci0 = &pci0;
+		pci1 = &pci1;
+		pci2 = &pci2;
+		usb0 = &usb0;
+		usb1 = &usb1;
+		dma0 = &dma0;
+		dma1 = &dma1;
+		sdhc = &sdhc;
+		msi0 = &msi0;
+		msi1 = &msi1;
+		msi2 = &msi2;
+
+		crypto = &crypto;
+		sec_jr0 = &sec_jr0;
+		sec_jr1 = &sec_jr1;
+		sec_jr2 = &sec_jr2;
+		sec_jr3 = &sec_jr3;
+		rtic_a = &rtic_a;
+		rtic_b = &rtic_b;
+		rtic_c = &rtic_c;
+		rtic_d = &rtic_d;
+		sec_mon = &sec_mon;
+
+		qman = &qman;
+		bman = &bman;
+		fman0 = &fman0;
+		fman1 = &fman1;
+	};
+
+/include/ "p4080si-chosen.dtsi"
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: PowerPC,e500mc@0 {
+			device_type = "cpu";
+			reg = <0>;
+			clocks = <&mux0>;
+			next-level-cache = <&L2_0>;
+			fsl,portid-mapping = <0x80000000>;
+			L2_0: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu1: PowerPC,e500mc@1 {
+			device_type = "cpu";
+			reg = <1>;
+			clocks = <&mux1>;
+			next-level-cache = <&L2_1>;
+			fsl,portid-mapping = <0x40000000>;
+			L2_1: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu2: PowerPC,e500mc@2 {
+			device_type = "cpu";
+			reg = <2>;
+			clocks = <&mux2>;
+			next-level-cache = <&L2_2>;
+			fsl,portid-mapping = <0x20000000>;
+			L2_2: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu3: PowerPC,e500mc@3 {
+			device_type = "cpu";
+			reg = <3>;
+			clocks = <&mux3>;
+			next-level-cache = <&L2_3>;
+			fsl,portid-mapping = <0x10000000>;
+			L2_3: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu4: PowerPC,e500mc@4 {
+			device_type = "cpu";
+			reg = <4>;
+			clocks = <&mux4>;
+			next-level-cache = <&L2_4>;
+			fsl,portid-mapping = <0x08000000>;
+			L2_4: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu5: PowerPC,e500mc@5 {
+			device_type = "cpu";
+			reg = <5>;
+			clocks = <&mux5>;
+			next-level-cache = <&L2_5>;
+			fsl,portid-mapping = <0x04000000>;
+			L2_5: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu6: PowerPC,e500mc@6 {
+			device_type = "cpu";
+			reg = <6>;
+			clocks = <&mux6>;
+			next-level-cache = <&L2_6>;
+			fsl,portid-mapping = <0x02000000>;
+			L2_6: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+		cpu7: PowerPC,e500mc@7 {
+			device_type = "cpu";
+			reg = <7>;
+			clocks = <&mux7>;
+			next-level-cache = <&L2_7>;
+			fsl,portid-mapping = <0x01000000>;
+			L2_7: l2-cache {
+				next-level-cache = <&cpc>;
+			};
+		};
+	};
+};
-- 
1.7.5.4

